tensorflow/core/kernels/data/captured_function.cc

```diff
@@ -450,8 +450,9 @@ Status MakeIteratorFromInputElement(
     std::unique_ptr<IteratorBase>* out_iterator) {
   std::vector<Tensor> return_values;
 
-  TF_RETURN_IF_ERROR(inst_captured_func.RunWithBorrowedArgs(ctx, input_element,
-                                                            &return_values));
+  TF_RETURN_IF_ERROR(inst_captured_func.RunWithBorrowedArgs(
```

Comment:
It would be preferable if we pipe a `Node` through `MakeIteratorFromInputElement` (to correctly account for processing time of transformations such as `flat_map` and `interleave`.

Reply:
Sounds good. By the way, the current implementation intentionally ignores functions outside the iterator's GetNext() path (e.g., Initialize code for Snapshot). Those are typically one-off operations and therefore I believe should not fall into processing time but let me know if you think otherwise.

Reply:
Using non-null Node only within GetNext makes sense. interleave and flat_map kernels end up calling the method within GetNext.

Reply:
Does this also require function overloading for internal API purposes?

Reply:
Yes, I was able to find internal calls to this API.